### ISSUE TITLE
"master and slave" database terminology should be on the word replacement list

### ISSUE THREAD

**Issue by pibion** [inclusivenaming/org#55]
It seems like there are replacements for "master" and "slave" database terminology that don't lose meaning.

Pilot and drone?  Primary and secondary?  Main and backup?

**Comment by dgrahn** [comment]
Primary and replica are commonly used.

- Pardon my brevity. I sent this from my phone.

**Comment by edwarnicke** [comment]
@pibion Thank you for bringing these up 'in context'.  One of the complex parts of moving to more inclusive language is its not a simple search/replace on the words.  Depending on the 'context' in which the words occur, there is often a more appropriate set for that context rooted in the that context.

For example, a lot of folks are using 'primary/secondary' as replacements for 'master/slave' ... and that's fantastic.  But while I'm not deep enough into storage to know what the correct words are in storage (and happy to believe folks like @dgrahn are)... 'primary/replica' seems reasonable to my ear.

@dgrahn Question... is 'replica' preferable in storage because often you have multiple 'replicas' ?

**Comment by dgrahn** [comment]
Yes. As I think about it, there should probably be multiple replacement
options since the secondary resources can fulfill multiple purposes.

Maybe before coming up with the replacement we need to nail down where the
contexts in which it is used so we can recommend appropriate replacements
in each case.

- Pardon my brevity. I sent this from my phone.

**Comment by edwarnicke** [comment]
@dgrahn I've always expressed that I felt word lists are just the first step on the journey.  We then need to get the right replacements agreed upon by the 'contexts'.  Different word replacements may fit for different 'contexts'.

**Comment by dgrahn** [comment]
Absolutely! Here's a few contexts I can think of. I'll use existing terminology here.

1. Master hosts a complete copy of the database. Slaves host copies / parts of copies. - alternative: source/replica?
2. Slaves hold parts of the database, master controls their behavior - alternative: controller/node?
3. Master is an authoritative version of an immutable dataset... Although that may be another term entirely.

Any to add?

**Comment by markcmiller86** [comment]
> Any to add?

Regarding `source/replica` ... does `replica` pass the plain language test? What about `original/copy` (that is often used in hard-copy copying business ;) For a non-plain-language option (😄 )`progenitor/inheritor`, in mathematical proofs ane expressions, `primal/derived`, C++ classes `base/derived`.

**Comment by edwarnicke** [comment]
@Nytelife26 you have a good background translatability... does ```replica``` in this context pass the plain language test to your you?

**Comment by dgrahn** [comment]
@edwarnicke Are these terms already on the word replacement list? (https://inclusivenaming.org/language/word-list/) Or is this issue for another list?

**Comment by Nytelife26** [comment]
> does `replica` in this context pass the plain language test

I believe so. It has equivalents in all languages to my knowledge.

Or, at least, it should, since it has been around since the 18th century
and was not originally an English word, but rather from romance
languages, and so it is likely it will have spread amongst all major
modern languages by now.

In Spanish at least, though, it is a cognate, so I can't weigh in for
other languages without researching further.

If not, copy or clone may potentially be viable and easier to understand.

**Comment by markcmiller86** [comment]
> I believe so. It has equivalents in all languages to my knowledge.

Not sure that is intended to answer the [*plane-language*](https://www.plainlanguage.gov/) question. My concern was whether *replica* is [well-enough known](https://datayze.com/word-analyzer?word=replica). That ref. suggests otherwise but I honestly didn't vet that resource at all. Nonetheless, I do believe we should be able to find high quality word scoring/ranking resources of similar purpose we can/should use to vet proposed replacements. And, for me anyways, direct translations in *all languages*, while admittedly useful for the reasons you've mentioned, is only one criteria among several.

**Comment by markcmiller86** [comment]
Capturing here the list of common replacements hosted on [Wikipedia](https://en.wikipedia.org/wiki/Master/slave_(technology))

**Comment by Nytelife26** [comment]
> My concern was whether *replica* is well-enough known.

I feel replica may be an overcomplication anyway. Ultimately, "copy" or
"clone" are far more widespread and carry the same meaning.

> Nonetheless, I do believe we should be able to find high quality word
> scoring/ranking resources of similar purpose we can/should use to vet
> proposed replacements.

Agreed. We should look into that.

> And, for me anyways, direct translations in *all
> languages*, while admittedly useful for the reasons you've mentioned, is
> only one criteria among several.

Oh, I know. In fact, I created a similar linguistics evaluation
framework in a publication I made, which you can see [here] if you're
interested.

[here]: https://github.com/kludge-cs/transparency/blob/main/open-statements/2021-04-04--TERMINOLOGY.md#a-framework-for-analysis

**Comment by keys-i** [comment]
Could we use the terms in bees like hive and drones or something similar which does carry plural meaning maybe we could use king and pawn. We should think about a term which describes the act master term does but is not necessarily misleading

**Comment by Nytelife26** [comment]
> Could we use the terms in bees like hive and drones [...] or king and pawn

My concern with these would be that they're still hierarchal rather than
based upon the relationship between the two entities, especially with the
latter relating to monarchal and feudal systems. On another note, it is not
immediately obvious what they mean, so I would argue it doesn't fit in with
plain language.

I would suggest something like control and copy, control and follow,
instructor and follower, etc. These have no relation to hierarchal systems
and they seem more immediately obvious to me.

Although, there are some other good replacements you can view here:
https://inclusivenaming.org/word-lists/tier-1/_master-slave

**Comment by dserodio** [comment]
The words **primary** and **replica** (or **standby**, depending on the context) are the most commonly used replacements.

Amazon Web Services' RDS and Google Cloud's Cloud SQL use **primary** and **replica**, and PostgreSQL calls them **primary** and **standby**.
